### DISCUSSION TITLE
Will the rnaspades.py filter short reads?

### DISCUSSION THREAD

**Discussion by Duannai** [ablab/spades#883] [Q&A]
Hello! When I read the manual of rnaspades.py, the rnaSPAdes uses 2 k-mer sizes, which are automatically detected using read length (approximately one third and half of the maximal read length). If the maximal read length is 150, the 2 k-mer sizes should be 50 and 75. So, will the reads shorter than 50 be automatically filtered?

Thanks in advance!

Mort

**Comment by andrewprzh** [comment]
Dear Mort,

Yes, reads shorter than k are ignored since they contain 0 k-mers.
During the first iteration SPAdes will use only reads of >50bp, and >75bp on the second iteration. You don't need to filter them manually.

Best
Andrey

**Reply by SarahIsme** [reply]
Sorry for another question.

For example, if the automatic k-mer sizes in rnaspades are 49 and 73, whether the results will be better when k-mer sizes are set as 49, 61, 73 (that is, adding a middle k-mer size) ?

Thanks in advance!

Sarah
